code/mob/living.dm

```diff
@@ -405,12 +405,16 @@
 			var/success = dropping_organ.attach_organ(src, dropping_user)
 			if (success)
 				return
-		else if (istype(dropped, /obj/item/parts/human_parts/))
-			var/obj/item/parts/dropping_limb = dropped
-			dropping_limb.attach(src, dropping_user)
-		else if (istype(dropped, /obj/item/parts/robot_parts/arm/) || istype(dropped, /obj/item/parts/robot_parts/leg/))
-			var/obj/item/parts/robot_parts/dropping_limb = dropped
-			dropping_limb.attack(src, dropping_user) // Attaching robot parts to humans is a bit complicated so we're going to be lazy and re-use attack.
+		else if (istype(dropped, /obj/item/parts))
+			if (istype(dropped, /obj/item/parts/human_parts/))
+				var/obj/item/parts/dropping_limb = dropped
+				dropping_limb.attach(src, dropping_user)
+			else if (istype(dropped, /obj/item/parts/robot_parts/arm/) || istype(dropped, /obj/item/parts/robot_parts/leg/))
+				var/obj/item/parts/robot_parts/dropping_limb = dropped
+				dropping_limb.attack(src, dropping_user) // Attaching robot parts to humans is a bit complicated so we're going to be lazy and re-use attack.
+			else if (istype(dropped, /obj/item/parts/artifact_parts))
+				var/obj/item/parts/artifact_parts/dropping_limb = dropped
+				dropping_limb.attach(src, dropping_user)
```

Comment:
I think this can be included in the first case, since the proc being called is the same? Assuming that 'attach' is defined on /obj/item/parts, just cast to that in the body and check `istype(human_parts) || istype(artifact_parts)` 

(Even if it isn't you can cheat and cast it to human_parts even if it's an artifact)

Reply:
Sure

Reply:
Actually artifact limbs have extra stuff defined in their attach() proc, didn't seem to work

Reply:
Nvm